spec.md

```diff
@@ -937,140 +783,208 @@ Property types are legal in the following constructs:
+When an inferred variant of an integer ground type is used, FIRRTL will calculate the minimum width needed for it.
+It is an error if the minimum size cannot be calculated.
 
-Two vector types are equivalent if they have the same length, and if their element types are equivalent.
+For example, the width of a multiplexer expression is the maximum of its two corresponding input widths.
 
-Two bundle types are equivalent if they have the same number of fields, and both the bundles' i'th fields have matching names and orientations, as well as equivalent types.
-Consequently, `{a:UInt, b:UInt}`{.firrtl} is not equivalent to `{b:UInt, a:UInt}`{.firrtl}, and `{a: {flip b:UInt}}`{.firrtl} is not equivalent to `{flip a: {b: UInt}}`{.firrtl}.
+The width of each primitive operation is detailed in [@sec:primitive-operations].
 
-Two property types are equivalent if they are the same concrete property type.
+### Reset Inference
+
+The uninferred `Reset`{.firrtl} type will be inferred to either a synchronous reset `UInt<1>`{.firrtl} or to an asynchronous reset `AsyncReset`{.firrtl}.
+
+The following example shows an inferred reset that will get inferred to a synchronous reset.
 
-## Width Inference
+``` firrtl
+input a : UInt<1>
+wire reset : Reset
+connect reset, a
+```
 
-For all circuit components declared with unspecified widths,
-the FIRRTL compiler will infer the minimum possible width that maintains the legality of all its incoming connections.
-If a circuit component has no incoming connections, and the width is unspecified, then an error is thrown to indicate that the width could not be inferred.
+After reset inference, `reset`{.firrtl} is inferred to the synchronous `UInt<1>`{.firrtl} type:
 
-For module input ports with unspecified widths, the inferred width is the minimum possible width that maintains the legality of all incoming connections to all instantiations of the module.
+``` firrtl
+input a : UInt<1>
+wire reset : UInt<1>
+connect reset, a
+```
 
-The width of a ground-typed multiplexer expression is the maximum of its two corresponding input widths.
-For multiplexing aggregate-typed expressions, the resulting widths of each leaf sub-element is the maximum of its corresponding two input leaf sub-element widths.
+The following example demonstrates usage of an asynchronous reset.
 
-The width of each primitive operation is detailed in [@sec:primitive-operations].
+``` firrtl
+input clock : Clock
+input reset : AsyncReset
+input x : UInt<8>
+regreset y : UInt<8>, clock, reset, UInt(123)
+; ...
+```
+
+Inference rules are as follows:
+
+1. An uninferred reset driven by and/or driving only asynchronous resets will be inferred as asynchronous reset.
+1. An uninferred reset driven by and/or driving both asynchronous and synchronous resets is an error.
+1. Otherwise, the reset is inferred as synchronous (i.e. the uninferred reset is only invalidated or is driven by or drives only synchronous resets).
```

Comment:
Mega-nit: 
```suggestion
1. Otherwise, the reset is inferred as synchronous. (I.e., the uninferred reset is only invalidated or is driven by or drives only synchronous resets.)
```

Reply:
This was the original copy.